tests/run-make/README.md

```diff
@@ -0,0 +1,35 @@
+# The `run-make` test suite
+
+The `run-make` test suite contain tests which are the most flexible out of all
+the [rust-lang/rust](https://github.com/rust-lang/rust) test suites. `run-make`
+tests can basically contain arbitrary code, and are supported by the
+[`run_make_support`] library.
+
+## Infrastructure
+
+The implementation for collecting and building the `rmake.rs` recipes (or
+`Makefile`s) are in
+[`src/tools/compiletest/src/runtest.rs`](https://github.com/rust-lang/rust/blob/master/src/tools/compiletest/src/runtest.rs),
+in `run_rmake_v2_test` and `run_rmake_legacy_test`.
+
+### Rust-based `run-make` tests: `rmake.rs`
+
+The setup for the `rmake.rs` version is a 3-stage process:
+
+1. First, we build the [`run_make_support`] library in bootstrap as a tool lib.
+2. Then, we compile the `rmake.rs` "recipe" linking the support library and its
+   dependencies in, and provide a bunch of env vars. We setup a directory
+   structure
+
+   ```
+   test_specific_build_dir/
```

Comment:
I would mention that this directory is inside `build/<target>/test`, so that people can find the directory and take a look at what's inside when a test fails.

Reply:
Added the `build/<target>/test/run-make/` path.